relayer/chains/cosmos/tx.go

```diff
@@ -691,6 +681,16 @@ func (cc *CosmosProvider) getTxFactoryAndBuilder(
 		txf = txf.WithFeeGranter(granterAddr)
 	}
 
+	adjusted := gas
+
+	if gas == 0 {
+		_, adjusted, err = cc.CalculateGas(ctx, txf, txSignerKey, cMsgs...)
+
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
```

Comment:
is it an issue with original relayer or our patch?

Reply:
can wait [PR](https://github.com/cosmos/relayer/pull/1364/files) for original get merged 1st